### PR TITLE
CLOUDP-134864: Envoy parses and sends proxy protocol TLV to upstream as HTTP headers

### PR DESCRIPTION
**Pull request by betsybutton**
[CLOUDP-134864](https://jira.mongodb.org/browse/CLOUDP-134864)

This PR contains the same changes as https://github.com/10gen/envoy-serverless/pull/6/files. I created a new PR to verify the commit containing the changes before merging.

Commit Message: Envoy parses and sends proxy protocol TLV to upstream as HTTP headers
Additional Description: Adds substitution formatter to access proxy protocol TLVs stored in the filter state 
Risk Level: Low
Testing: Manually tested and unit test added
Docs Changes: Should be added to list of available substitution formatters when merged upstream

### CODE REVIEW

**Review by visualzhou — APPROVED** [review]
LGTM. I wonder why you chose to make a new PR than updating #6. If signature was the only reason, you could force push the branch. That will lose previous commits and made comments on them out of date though.

**Comment by betsybutton** [comment]
@visualzhou I assumed this was the best practice because it's the method recommended on [our wiki](https://wiki.corp.mongodb.com/display/MMS/Setup+Git+Commit+Signing#SetupGitCommitSigning-HELP,I'vepushedunsignedcommits). Thanks for the 2nd LGTM!

File: source/common/formatter/substitution_formatter.cc [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #include <cstdlib>
 #include <memory>
 #include <regex>
+#include <stdexcept>
```
Comment (visualzhou) [inline source/common/formatter/substitution_formatter.cc:10]: @betsybutton  I realized this is an unused include after the refactoring. Could you please put a quick fix on the same ticket for this?

Comment (betsybutton) [reply source/common/formatter/substitution_formatter.cc:10]: Oh, nice catch! Absolutely.

Comment (betsybutton) [reply source/common/formatter/substitution_formatter.cc:10]: @visualzhou Here's the PR: https://github.com/10gen/envoy-serverless/pull/17

**Pull request merged**